### PR TITLE
docs(readme): added types

### PR DESCRIPTION
**Pull request by harish-sethuraman**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
Changed to typescript types and standardized readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -306,7 +319,12 @@ More [`examples`](#examples)
 
 #### `globOptions`
 
-Type: `Object`
+Type:
+
+```ts
+type globOptions = object;
```
Comment (alexander-akait) [inline README.md:325]: Let's use types from `globby`, just use the same name

File: README.md [hunk 2]
```diff
@@ -335,7 +353,12 @@ module.exports = {
 
 #### `filter`
 
-Type: `Function`
+Type:
+
+```ts
+type filter = (filepath: string) => any;
```
Comment (alexander-akait) [inline README.md:359]: Should return boolean

File: README.md [hunk 3]
```diff
@@ -370,19 +393,24 @@ module.exports = {
 
 #### `toType`
 
-Type: `String`
+Type:
+
+```ts
+type toType = string;
```
Comment (alexander-akait) [inline README.md:399]: It is literal types, so better to use `"dir" | "file" | "other-values"`

File: README.md [hunk 4]
```diff
@@ -506,12 +544,17 @@ module.exports = {
 
 #### `transform`
 
-Type: `Function|Object`
+Type:
+
+```ts
+type transform = object | ((input: string, absoluteFilename: string) => string);
```
Comment (alexander-akait) [inline README.md:550]: Expand `object`

File: README.md [hunk 5]
```diff
@@ -595,15 +643,20 @@ module.exports = {
 
 ###### `cache`
 
-Type: `Boolean|Object`
+Type:
+
+```ts
+type cache = boolean | object;
```
Comment (alexander-akait) [inline README.md:649]: Expand `object`

File: README.md [hunk 6]
```diff
@@ -738,7 +791,18 @@ module.exports = {
...
+Type:
+
+```ts
+type transformAll = (
+  data: {
+    data: Buffer;
+    sourceFilename: string;
+    absoluteFilename: string;
+  }[]
+) => any;
```
Comment (alexander-akait) [inline README.md:803]: Should not be `any`

File: README.md [hunk 7]
```diff
@@ -799,7 +868,12 @@ module.exports = {
 
 #### `info`
 
-Type: `Object|Function<Object>`
+Type:
+
+```ts
+type info = object | (() => object);
```
Comment (alexander-akait) [inline README.md:874]: Better to use `Recond<string, any>`

File: README.md (hunk 2 above)
Comment (harish-sethuraman) [reply README.md:359]: - https://github.com/webpack-contrib/copy-webpack-plugin/blob/9b5a314b71e9ab16ad3ada0816620aa75d18ca47/types/index.d.ts#L252

I took it from here?

File: README.md (hunk 6 above)
Comment (harish-sethuraman) [reply README.md:803]: - https://github.com/webpack-contrib/copy-webpack-plugin/blob/9b5a314b71e9ab16ad3ada0816620aa75d18ca47/types/index.d.ts#L70

🤔  Am I looking at the wrong type then 😕

File: README.md (hunk 2 above)
Comment (alexander-akait) [reply README.md:359]: Let's fix our types too :smile:

Comment (harish-sethuraman) [reply README.md:359]: Agreed

File: README.md (hunk 6 above)
Comment (alexander-akait) [reply README.md:803]: We should add `@returns` to types here too, looks like bug on our side

Comment (harish-sethuraman) [reply README.md:803]: - https://github.com/webpack-contrib/copy-webpack-plugin/blob/9b5a314b71e9ab16ad3ada0816620aa75d18ca47/types/index.d.ts#L253-L259

I thought this returned any

File: README.md (hunk 1 above)
Comment (harish-sethuraman) [reply README.md:325]: import("globby").Options?

File: types/index.d.ts [hunk 8]
```diff
@@ -249,7 +249,7 @@ type TransformerObject = {
   cache?: boolean | TransformerCacheObject | undefined;
 };
 type Transform = TransformerFunction | TransformerObject;
-type Filter = (filepath: string) => any;
+type Filter = (filepath: string) => boolean;
```
Comment (alexander-akait) [inline types/index.d.ts:252]: We should add this here https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/index.js#L95 and run `npm run build`, same for https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/index.js#L99

Comment (harish-sethuraman) [reply types/index.d.ts:254]: Done

File: types/index.d.ts [hunk 9]
```diff
@@ -64,10 +64,12 @@ export = CopyPlugin;
 /**
  * @callback Filter
  * @param {string} filepath
+ * @returns {boolean}
  */
 /**
  * @callback TransformAllFunction
  * @param {{ data: Buffer, sourceFilename: string, absoluteFilename: string }[]} data
+ * @returns {string}
```
Comment (alexander-akait) [inline types/index.d.ts:72]: Data can be `string | Buffer`

Comment (harish-sethuraman) [reply types/index.d.ts]: Done

**Comment by alexander-akait** [comment]
Thanks

**Pull request merged**